iwyu.cc

```diff
@@ -2041,9 +2036,7 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     // We always delete a pointer, so do one dereference to get the
     // actual type being deleted.
     const Type* delete_ptr_type = GetTypeOf(delete_arg);
-    const Type* delete_type = RemovePointerFromType(delete_ptr_type);
-    if (delete_type && !IsPointerOrReferenceAsWritten(delete_type))
-      ReportTypeUse(CurrentLoc(), delete_type);
+    ReportTypeUse(CurrentLoc(), delete_ptr_type, UseKind::RemoveRefsAndPtr);
```

Comment:
Curious: why `RemoveRefsAndPtr` here? Is to cover scenarios like:
```
T* p = new T;
T& r = *p;
delete &r;
```
?

Reply:
It's rather for:
```cpp
T* p = new T;
T*& r = p;
delete r;
```
And there may be additional references through type aliases.